compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -339,8 +339,15 @@ impl<'a> Linker for GccLinker<'a> {
     }
 
     fn link_dylib(&mut self, lib: Symbol) {
+        let larg = format!("-l{}", lib);
+        if self.sess.target.os == "illumos" && larg == "-lc" {
```

Comment:
```suggestion
        if self.sess.target.os == "illumos" && lib == "c" {
```
(or `lib.as_str() == "c"` if the former doesn't compile)